source/common/http/http2/codec_impl.cc

```diff
@@ -951,7 +951,12 @@ int ConnectionImpl::onInvalidFrame(int32_t stream_id, int error_code) {
     stream->setDetails(Http2ResponseCodeDetails::get().errorDetails(error_code));
   }
 
-  if (error_code == NGHTTP2_ERR_HTTP_HEADER || error_code == NGHTTP2_ERR_HTTP_MESSAGING) {
+  switch (error_code) {
+  case NGHTTP2_ERR_REFUSED_STREAM:
+    return 0;
+
+  case NGHTTP2_ERR_HTTP_HEADER:
+  case NGHTTP2_ERR_HTTP_MESSAGING: {
```

Comment:
It would be good to more carefully enumerate error codes that we expect to get past this code and codes that we do not expect to run into.  One example for a code that we probably never want to see is `NGHTTP2_ERR_STREAM_CLOSED`

Reply:
I only found `NGHTTP2_ERR_FLOW_CONTROL` and `NGHTTP2_ERR_PROTO` additionally that are being sent via this path. Is `NGHTTP2_ERR_STREAM_CLOSED` being thrown additionally from elsewhere?